File: game.py
import random
import time

def play_round():
    """
    Function to play a single round of Rock, Paper, Scissors.
    """
    num = random.randrange(1, 99)
    if num <= 33:
        comp = 'r'
    elif num >= 66:
        comp = 's'
    else:
        comp = 'p'

    print("Enter (r)rock, (p)papers, or (s)scissors:")
    while True:
        user_input = input().lower()  # Convert input to lowercase for case-insensitivity
        if user_input in ['r', 'p', 's']:
            break
        else:
            print("Invalid input! Please try again")

    print("Thinking...")
    time.sleep(1)

    if user_input == comp:
        print("Match draw")
    elif (user_input == 'r' and comp == 's') or (user_input == 'p' and comp == 'r') or (user_input == 's' and comp == 'p'):
        print("Congratulations! You won")
    else:
        print("You lost! Better luck next time")

def main():
    """
    Main function to control the flow of the game.
    """
    loop_condition = True

    while loop_condition:
        play_round()
        
        print("Wanna play again? (y/n)")
        while True:
            end_input = input().lower()  # Convert input to lowercase for case-insensitivity
            if end_input == 'n':
                loop_condition = False
            elif end_input == 'y':
                break
            else:
                print("Invalid input! Please try again")

if __name__ == "__main__":
    main()

File: game.js
function getComputerChoice() {
    const num = Math.floor(Math.random() * 99) + 1;
    if (num <= 33)
        return 'r';
    else if (num >= 66)
        return 's';
    else
        return 'p';
}

function playRound() {
    const comp = getComputerChoice();
    let user_input;
    while (true) {
        user_input = prompt("Enter (r)rock, (p)papers, or (s)scissors:");
        if (user_input === 'r' || user_input === 'p' || user_input === 's')
            break;
        else
            alert("Invalid input! Please try again");
    }
    alert("Thinking...");
    // Simulating thinking time
    for (let i = 0; i < 1000000000; i++);
    
    if (user_input === comp)
        alert("Match draw");
    else if ((user_input === 'r' && comp === 's') || (user_input === 'p' && comp === 'r') || (user_input === 's' && comp === 'p'))
        alert("Congratulations! You won");
    else
        alert("You lost! Better luck next time");
}

function main() {
    let loop_condition = true;

    while (loop_condition) {
        playRound();
        
        let end_input;
        while (true) {
            end_input = prompt("Wanna play again? (y/n)").toLowerCase();
            if (end_input === 'n') {
                loop_condition = false;
                break;
            } else if (end_input === 'y') {
                break;
            } else {
                alert("Invalid input! Please try again");
            }
        }
    }
}

main();